src/lib.rs

```diff
@@ -423,76 +423,166 @@ fn test(doc: &Value, path: &str, expected: &Value) -> Result<(), PatchErrorKind>
 /// # }
 /// ```
 pub fn patch(doc: &mut Value, patch: &[PatchOperation]) -> Result<(), PatchError> {
-    apply_patches(doc, 0, patch)
+    let mut undo_stack = Vec::new();
```

Comment:
Really minor thing -- do you think you can reserve undo stack size? `Vec::with_capacity(patch.len())`?